### DISCUSSION TITLE
see all data_vars

### DISCUSSION THREAD

**Discussion by raybellwaves** [pydata/xarray#4991] [Q&A] [answered]
the data_vars method provides a convenient ways to see the varaibles and the formatting (https://github.com/pydata/xarray/blob/master/xarray/core/dataset.py#L484) is nice. However, there are times I would look to see the full variables names not just

```
Data variables: (12/N)
    a  (y, x) float32 ...
    b         (y, x) float32 ...
    c        (y, x) float32 ...
    d          (y, x) float32 ...
    e      (y, x) float32 ...
    f        (y, x) float32 ...
    ...         ...
    g      (y, x) float32 ...
    h       (y, x) float32 ...
    i      (y, x) float32 ...
    j      (y, x) float32 ...
    k      (y, x) float32 ...
    l     (y, x) float32 ...
``` 

Doing `.variables` will give me the cords as well and some data.

Ideally there would be a setting to toggle to specify the length of the formatter akin to https://pandas.pydata.org/pandas-docs/stable/user_guide/options.html#overview

`xr.set_option('display.max_data_vars', N)`

**Comment by shoyer** [comment, marked as answer]
I agree, this would be a welcome option!

In the meantime, you might try something like `print(', '.join(ds.data_vars))` for printing the full list of data variables.

**Comment by raybellwaves** [comment]
I also just learned this is available `xr.set_options(display_max_rows=20)` (https://github.com/pydata/xarray/issues/4992)
